lib/Driver/CompilationRecord.h

```diff
@@ -0,0 +1,51 @@
+//===--- CompilationRecord.h ------------------------------------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_DRIVER_COMPILATIONRECORD_H
+#define SWIFT_DRIVER_COMPILATIONRECORD_H
+
+namespace swift {
+namespace driver {
+namespace compilation_record {
+
+/// Compilation record files (.swiftdeps files) are YAML files composed of these
+/// top-level keys.
+enum class TopLevelKey {
+  /// The key for the Swift compiler version used to produce the compilation
+  /// record.
+  Version,
+  /// The key for the list of arguments passed to the Swift compiler when
+  /// producing the compilation record.
+  Options,
+  /// The key for the time at which the build that produced the compilation
+  /// record started.
+  BuildTime,
+  /// The key for the list of inputs to the compilation that produced the
+  /// compilation record.
+  Inputs,
+};
+
+/// \returns A string representation of the given key.
+inline StringRef getName(TopLevelKey Key) {
```

Comment:
Please mark this `static`. Non-static inlines have weird rules that we just avoid.